### PR TITLE
Enable discoverable devices probe kselftest on MediaTek chromebooks

### LINKED ISSUE #579: Enable 'discoverable devices probe' kselftest on MediaTek Chromebooks
This task is to track progress on enabling the [discoverable devices probe kselftest](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=dacf1d7a78bf8a131346c47bfba7fe1f3ff44beb) on all MediaTek Chromebooks.

### PR DESCRIPTION
**Pull request by nfraprado**
Enable the discoverable devices probe kselftest on the MediaTek Chromebook platforms. This test will validate that the devices connected through USB or PCI on those platforms have been probed successfully. The devices are, namely, the camera on mt8183-juniper and mt8186-steelix, and the camera, wifi and bluetooth on mt8195-tomato.

Depends on https://github.com/kernelci/kernelci-core/pull/2577.

Fixes #579 .

### CODE REVIEW

**Comment by nfraprado** [comment]
Manual LAVA runs for reference:
mt8195-tomato: https://lava.collabora.dev/scheduler/job/14310494
mt8183-juniper: https://lava.collabora.dev/scheduler/job/14310955
mt8186-steelix: https://lava.collabora.dev/scheduler/job/14310717

File: config/pipeline.yaml [hunk 1]
```diff
@@ -447,6 +447,20 @@ jobs:
         patchlevel: 7
     kcidb_test_suite: kselftest.dt
 
+  kselftest-discoverable-devices-probe:
+    template: kselftest.jinja2
```
Comment (pawiecz) [inline config/pipeline.yaml:451]: Depending on when this PR gets merged, be advised that the job template might need some adjustment; see https://github.com/kernelci/kernelci-pipeline/pull/655

File: config/pipeline.yaml [hunk 2]
```diff
@@ -447,6 +447,20 @@ jobs:
...
+    kind: test
```
Comment (pawiecz) [inline config/pipeline.yaml:452]: Is this a single test case or a test suite? It [might need adjusting its `kind`](https://github.com/kernelci/kernelci-pipeline/pull/648#discussion_r1665524836); see https://github.com/kernelci/kernelci-pipeline/commit/396c67d0ba5260a56b4c4ceccd1635665150f527

File: config/pipeline.yaml [hunk 3]
```diff
@@ -447,6 +447,20 @@ jobs:
...
+    params:
+      nfsroot: 'http://storage.kernelci.org/images/rootfs/debian/bookworm-kselftest/20240221.0/{debarch}' # TODO: update once new rootfs with platform-test-parameters is generated
```
Comment (pawiecz) [inline config/pipeline.yaml:454]: Try https://storage.staging.kernelci.org/rootfs/bookworm-kselftest/20240704.0 (just to spin this up on staging)

File: config/scheduler-chromeos.yaml [hunk 4]
```diff
@@ -187,6 +187,14 @@ scheduler:
       result: pass
     platforms: *mediatek-platforms
 
+  - job: kselftest-discoverable-devices-probe
+    <<: *lava-job-collabora
+    event:
+      channel: node
+      name: kbuild-gcc-10-arm64-chromebook
```
Comment (pawiecz) [inline config/scheduler-chromeos.yaml:194]: GCC was recently bumped to 12; see https://github.com/kernelci/kernelci-pipeline/commit/ac15e371e2d5cf8bd1bdc621493892346658f0e6

**Review by pawiecz — COMMENTED** [review]
LGTM (left a few minor comments)

File: config/scheduler-chromeos.yaml [hunk 5]
```diff
@@ -187,6 +187,14 @@ scheduler:
       result: pass
     platforms: *mediatek-platforms
 
+  - job: kselftest-discoverable-devices-probe
```
Comment (musamaanjum) [inline config/scheduler-chromeos.yaml:190]: Is `discoverable-devices-probe` a kselftest? If not, why the same naming layout is being used?

File: config/pipeline.yaml (hunk 2 above)
Comment (JenySadadia) [reply config/pipeline.yaml:452]: I agree with @pawiecz's suggestion. It should have `kind: job` in case of a test suite.

Comment (nfraprado) [reply config/pipeline.yaml]: It is a test suite. Fixed.

File: config/pipeline.yaml (hunk 3 above)
Comment (nfraprado) [reply config/pipeline.yaml]: Done. Leaving the comment so I remember to update it to a production URL before merging.

File: config/scheduler-chromeos.yaml (hunk 4 above)
Comment (nfraprado) [reply config/scheduler-chromeos.yaml]: Fixed.

File: config/scheduler-chromeos.yaml (hunk 5 above)
Comment (nfraprado) [reply config/scheduler-chromeos.yaml:204]: It is a kselftest: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=dacf1d7a78bf8a131346c47bfba7fe1f3ff44beb

**Comment by nfraprado** [comment]
Since the dependencies are available in staging I've removed staging-skip label.

**Comment by nfraprado** [comment]
I just noticed that I forgot to document that this change also depends on a kernel series: https://lore.kernel.org/all/20240613-kselftest-discoverable-probe-mt8195-kci-v1-0-7b396a9b032d@collabora.com

It is present on linux-next, so I've updated the `rules` for the job accordingly so the PR can already be tested and will run only on next. Also added a comment about how to change the rule in ~3 weeks from now to start running this test on mainline as well.

File: config/pipeline.yaml [hunk 6]
```diff
@@ -857,6 +857,21 @@ jobs:
         patchlevel: 7
     kcidb_test_suite: kselftest.dt
 
+  kselftest-discoverable-devices-probe:
```
Comment (musamaanjum) [inline config/pipeline.yaml:860]: We usually use only the name of the test suite at the end of the job name. Maybe better to follow the same naming scheme and rename it to just `kselftest-discoverable-devices` as suite name is `devices`. The same rename should be done for kcidb_test_suite property.

Comment (nfraprado) [reply config/pipeline.yaml]: Updated the name to match the kselftest collection like the others.

**Comment by nfraprado** [comment]
I've updated the kselftest collection name to `devices/probe` following the move in https://lore.kernel.org/all/20240705-dev-err-log-selftest-v2-1-163b9cd7b3c1@collabora.com

File: config/pipeline.yaml (hunk 1 above)
Comment (pawiecz) [reply config/pipeline.yaml:861]: This job template no longer exists. It was replaced by [`generic.jinja2`](https://github.com/kernelci/kernelci-pipeline/blob/main/config/runtime/generic.jinja2) - see [the example](https://github.com/kernelci/kernelci-pipeline/blob/main/doc/developer-documentation.md#an-example-of-enabling-a-new-job) for more details.

Comment (nfraprado) [reply config/pipeline.yaml]: Updated.

**Comment by nfraprado** [comment]
I've generated the rootfs on staging for https://github.com/kernelci/kernelci-core/pull/2577 and updated the URL here, so removing the staging-skip tag so this can get tested.

**Comment by nfraprado** [comment]
Tested on all platforms.

MT8186 OK:
https://lava.collabora.dev/scheduler/job/15670426

MT8183 OK:
https://lava.collabora.dev/scheduler/job/15672968

MT8192 not OK:
https://lava.collabora.dev/scheduler/job/15672970
The board file was not present in repo. Now merged through https://github.com/kernelci/platform-test-parameters/pull/2.

MT8195 partial OK:
https://lava.collabora.dev/scheduler/job/15672971
USB devices were failing [due to unmerged PR](https://github.com/kernelci/platform-test-parameters/pull/1), has been merged now.

So once https://github.com/kernelci/kernelci-core/pull/2577 is merged and the rootfs available in the production storage, its URL can be updated here and this PR merged.

**Comment by nuclearcat** [comment]
Can you please rebase and let's merge?

**Comment by nfraprado** [comment]
@nuclearcat I also need to update the rootfs URL to the production rootfs URL for bookworm-kselftest, but the latest in https://storage.kernelci.org/images/rootfs/debian/bookworm-kselftest/ is still `20240313.0/`, so it seems that the rootfs hasn't been generated yet with https://github.com/kernelci/kernelci-core/pull/2577 merged. Does the production rootfs generation need to be done manually?

**Comment by musamaanjum** [comment]
@nfraprado we have CI for this here (https://github.com/kernelci/kernelci-core/actions/workflows/rootfs.yml). Not sure if you require permissions to run it. 

Click on "Run Workflow". Select the branch, rootfs name, architecture and destination. The destination must be staging for verification of rootfs. Once you verify the results of your test on staging, create the rootfs on production and then merge the PR.

**Comment by nfraprado** [comment]
Hm, I just ran it here: https://github.com/kernelci/kernelci-core/actions/runs/11165276908/job/31036570425

But still don't see it in https://storage.staging.kernelci.org/images/rootfs/debian/bookworm-kselftest/

**Comment by musamaanjum** [comment]
Screenshot from 2022-10-05 13-10-23
I've created a job (https://github.com/kernelci/kernelci-core/actions/runs/11176504665) as well with the parameters mentioned in the above image. I'm unable to find the artifacts on storage.staging.

I'll leave it and let it get debugged by original authors.

**Comment by nuclearcat** [comment]
https://storage.staging.kernelci.org/rootfs/bookworm-kselftest/

**Comment by nfraprado** [comment]
Tested OK: https://lava.collabora.dev/scheduler/job/15963263

So I've deployed the rootfs to production: https://github.com/kernelci/kernelci-core/actions/runs/11220957370

And updated the PR here to use the proper URL for the production rootfs. So now it's ready to merge.

**Pull request merged**